mapr/final.sls

```diff
@@ -214,57 +234,105 @@ login:
     - require:
       - cmd: add-password
 
-# Give oozie time to spin up
+# Give things time to spin up
 wait:
   cmd:
     - run
     - name: sleep 30
     - require:
       - cmd: login
 
-{% if pillar.mapr.encrypted %}
 
-stop-oozie:
+{% if 'mapr.yarn.resourcemanager' in grains.roles %}
+
+# Restart the RM to make sure it picks up the extra config in yarn-site
+restart-rm:
   cmd:
     - run
-    - name: 'maprcli node services -name oozie -action stop -nodes {{ grains.fqdn }}'
+    - name: 'maprcli node services -name resourcemanager -action restart -nodes {{ grains.fqdn }}'
     - user: mapr
-    - onlyif: test -f /opt/mapr/roles/oozie
     - require:
       - cmd: login
       - cmd: wait
+      - file: yarn-site
+    - require_in:
+      - cmd: logout
 
-oozie-secure-war:
+{% endif %}
+
+
+{% if 'mapr.mapreduce.historyserver' in grains.roles %}
+
+# Restart the HS to make sure it picks up the extra config in yarn-site
+restart-hs:
   cmd:
     - run
-    - name: '/opt/mapr/oozie/oozie-4.2.0/bin/oozie-setup.sh -hadoop 2.7.0 /opt/ -secure'
+    - name: 'maprcli node services -name historyserver -action restart -nodes {{ grains.fqdn }}'
     - user: mapr
-    - onlyif: test -f /opt/mapr/roles/oozie
     - require:
-      - cmd: stop-oozie
+      - cmd: login
+      - cmd: wait
+      - file: yarn-site
+    - require_in:
+      - cmd: logout
 
-start-oozie:
+{% endif %}
+
+
+{% if 'mapr.yarn.nodemanager' in grains.roles %}
+
+# Wait again to make sure the resourcemanager gets a chance to restart first
+wait-nm:
   cmd:
     - run
-    - name: 'maprcli node services -name oozie -action start -nodes {{ grains.fqdn }}'
+    - name: sleep 30
+    - require:
+      - cmd: login
+
+# Restart the NM to make sure it picks up the extra config in yarn-site
+restart-nm:
+  cmd:
+    - run
+    - name: 'maprcli node services -name nodemanager -action restart -nodes {{ grains.fqdn }}'
     - user: mapr
-    - onlyif: test -f /opt/mapr/roles/oozie
     - require:
-      - cmd: oozie-secure-war
+      - cmd: login
+      - cmd: wait
+      - cmd: wait-nm
+      - file: yarn-site
     - require_in:
       - cmd: logout
 
-{% else %}
+{% endif %}
 
-restart-oozie:
+
+{% if 'mapr.oozie' in grains.roles and pillar.mapr.encrypted %}
+
+stop-oozie:
   cmd:
     - run
-    - name: 'maprcli node services -name oozie -action restart -nodes {{ grains.fqdn }}'
+    - name: 'maprcli node services -name oozie -action stop -nodes {{ grains.fqdn }}'
     - user: mapr
-    - onlyif: test -f /opt/mapr/roles/oozie
     - require:
       - cmd: login
       - cmd: wait
+
+oozie-secure-war:
+  cmd:
+    - run
+    - name: '/opt/mapr/oozie/oozie-4.2.0/bin/oozie-setup.sh -hadoop 2.7.0 /opt/mapr/hadoop/hadoop-2.7.0 -secure'
```

Comment:
Ahh ok makes sense.  Yeah if there's some way to get that info that seems beneficial, otherwise it sounds like we need to note that this only works with a specific version of MapR.